### ISSUE TITLE
No DHCP ACK for Unifi Access point

### ISSUE THREAD

**Issue by tebrevik** [systemd/systemd#22513]
**systemd version the issue has been seen with**
 > systemd-networkd-249.9-1.fc35.x86_64

**Used distribution**
 > Fedora 35

**Linux kernel version used** (`uname -a`)

 > `Linux fedora 5.16.7-200.fc35.x86_64 #1 SMP PREEMPT Sun Feb 6 19:53:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > Intel(R) Xeon(R) CPU E31260L (x86_64)

**Expected behaviour you didn't see**
 > Expected to see ACK after REQUEST
 like this:
```
eno2: DHCPv4 server: DISCOVER (0x894f0c7a)
eno2: DHCPv4 server: OFFER (0x894f0c7a)
eno2: DHCPv4 server: REQUEST (selecting) (0x894f0c7a)
eno2: DHCPv4 server: ACK (0x894f0c7a)
```

**Unexpected behaviour you saw**
 > Repeated DISCOVER, OFFER AND REQUEST, no ACK
```
eno2: DHCPv4 server: DISCOVER (0x719e30fe)
eno2: DHCPv4 server: OFFER (0x719e30fe)
eno2: DHCPv4 server: REQUEST (selecting) (0x719e30fe)
eno2: NDISC: Sent Router Solicitation, next solicitation in 29s
eno2: DHCPv4 server: REQUEST (selecting) (0x719e30fe)
eno2: DHCPv4 server: REQUEST (selecting) (0x719e30fe)
rtnl: received non-static neighbor, ignoring.
eno2: LLDP: Invoking callback for 'updated' event.
eno2: LLDP: Successfully processed LLDP datagram.
eno2: DHCPv4 server: DISCOVER (0x1245aaf8)
eno2: DHCPv4 server: OFFER (0x1245aaf8)
eno2: DHCPv4 server: REQUEST (selecting) (0x1245aaf8)
eno2: DHCPv4 server: REQUEST (selecting) (0x1245aaf8)
eno1: LLDP: Invoking callback for 'refreshed' event.
eno1: LLDP: Successfully processed LLDP datagram.
eno2: DHCPv4 server: REQUEST (selecting) (0x1245aaf8)
rtnl: received non-static neighbor, ignoring.
eno2: NDISC: Sent Router Solicitation, next solicitation in 58s
eno2: DHCPv4 server: DISCOVER (0x35c41cc4)
eno2: DHCPv4 server: OFFER (0x35c41cc4)
eno2: DHCPv4 server: REQUEST (selecting) (0x35c41cc4)
eno2: DHCPv4 server: REQUEST (selecting) (0x35c41cc4)
```

**Steps to reproduce the problem**
 > Connect a Ubiquiti Unifi AP to the network (tested with uap-ac-lr and uap-ac-lite). My assumption has been that it is something with the unifi ap's, but I do not see any output in the debug output for systemd-networkd which indicate that there is a problem with the REQUEST.

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
Snippet from tcpdump:

23:27:07.840372 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)               
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x7affd0ae, secs 86, Flags [none] (0x0000)              
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Discover                                                                                                                             
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f                                                                                                                 
            Requested-IP (50), length 4: 192.168.57.2                                                                                                                         
            MSZ (57), length 2: 576                                                                                                                                           
            Parameter-Request (55), length 8:                                                                                                                                 
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)                                                                                     
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)                                                                                                         
            Vendor-Class (60), length 4: "ubnt"                                                                                                                               
23:27:07.840480 bc:30:5b:e7:38:92 > fc:ec:da:89:71:5f, ethertype IPv4 (0x0800), length 326: (tos 0xc0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 312)      
    192.168.57.1.bootps > 192.168.57.2.bootpc: [udp sum ok] BOOTP/DHCP, Reply, length 284, xid 0x7affd0ae, Flags [none] (0x0000)                                              
          Your-IP 192.168.57.2                                                                                                                                                
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Offer                                                                                                                                
            Lease-Time (51), length 4: 3600                                                                                                                                   
            Subnet-Mask (1), length 4: 255.255.255.0                                                                                                                          
            Default-Gateway (3), length 4: 192.168.57.1                                                                                                                       
            Domain-Name (15), length 4: "test"                                                                                                                              
            Vendor-Option (43), length 6: 1.4.192.168.56.188                                                                                                                  
            Server-ID (54), length 4: 192.168.57.1                                                                                                                            
23:27:07.840487 bc:30:5b:e7:38:92 > fc:ec:da:89:71:5f, ethertype IPv4 (0x0800), length 326: (tos 0xc0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 312)      
    192.168.57.1.bootps > 192.168.57.2.bootpc: [udp sum ok] BOOTP/DHCP, Reply, length 284, xid 0x7affd0ae, Flags [none] (0x0000)                                              
          Your-IP 192.168.57.2                                                                                                                                                
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Offer                                                                                                                                
            Lease-Time (51), length 4: 3600                                                                                                                                   
            Subnet-Mask (1), length 4: 255.255.255.0                                                                                                                          
            Default-Gateway (3), length 4: 192.168.57.1                                                                                                                       
            Domain-Name (15), length 4: "test"                                                                                                                              
            Vendor-Option (43), length 6: 1.4.192.168.56.188                                                                                                                  
            Server-ID (54), length 4: 192.168.57.1                                                                                                                            
23:27:07.841026 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)               
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x7affd0ae, secs 86, Flags [none] (0x0000)              
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Request                                                                                                                              
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f                                                                                                                 
            Requested-IP (50), length 4: 192.168.57.2                                                                                                                         
            Server-ID (54), length 4: 192.168.57.1                                                                                                                            
            MSZ (57), length 2: 576                                                                                                                                           
            Parameter-Request (55), length 8:                                                                                                                                 
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)                                                                                     
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)                                                                                                         
            Vendor-Class (60), length 4: "ubnt"                                                                                                                               
23:27:10.844451 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)               
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x7affd0ae, secs 89, Flags [none] (0x0000)              
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Request                                                                                                                              
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f                                                                                                                 
            Requested-IP (50), length 4: 192.168.57.2                                                                                                                         
            Server-ID (54), length 4: 192.168.57.1                                                                                                                            
            MSZ (57), length 2: 576                                                                                                                                           
            Parameter-Request (55), length 8:                                                                                                                                 
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)                                                                                     
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)                                                                                                         
            Vendor-Class (60), length 4: "ubnt"                                                                                                                               
23:27:13.847698 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)               
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x7affd0ae, secs 92, Flags [none] (0x0000)              
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Request                                                                                                                              
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f                                                                                                                 
            Requested-IP (50), length 4: 192.168.57.2                                                                                                                         
            Server-ID (54), length 4: 192.168.57.1                                                                                                                            
            MSZ (57), length 2: 576                                                                                                                                           
            Parameter-Request (55), length 8:                                                                                                                                 
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)                                                                                     
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)                                                                                                         
            Vendor-Class (60), length 4: "ubnt"                                                                                                                               
23:27:27.086413 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)               
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0xf216f80, secs 105, Flags [none] (0x0000)              
          Client-Ethernet-Address fc:ec:da:89:71:5f                                                                                                                           
          Vendor-rfc1048 Extensions                                                                                                                                           
            Magic Cookie 0x63825363                                                                                                                                           
            DHCP-Message (53), length 1: Discover                                                                                                                             
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f                                                                                                                 
            Requested-IP (50), length 4: 192.168.57.2                                                                                                                         
            MSZ (57), length 2: 576                                                                                                                                           
            Parameter-Request (55), length 8:                                                                                                                                 
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)                                                                                     
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)                                                                                                         
            Vendor-Class (60), length 4: "ubnt"                                                                                                                               
```

**Comment by yuwata** [comment]
@tebrevik Could you show the packets in hex, that is, use -x option for tcpdump?

**Comment by tebrevik** [comment]
> @tebrevik Could you also show the packets in hex, that is, use `-x -vvv` option for tcpdump?

sure, here it is:

```
tcpdump: listening on eno2, link-type EN10MB (Ethernet), snapshot length 262144 bytes
21:23:44.288323 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x1e35286a, secs 3, Flags [none] (0x0000)
          Client-Ethernet-Address fc:ec:da:89:71:5f
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Discover
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f
            MSZ (57), length 2: 576
            Parameter-Request (55), length 8:
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)
            Vendor-Class (60), length 4: "ubnt"
        0x0000:  4500 0148 0000 0000 4011 79a6 0000 0000
        0x0010:  ffff ffff 0044 0043 0134 580e 0101 0600
        0x0020:  1e35 286a 0003 0000 0000 0000 0000 0000
        0x0030:  0000 0000 0000 0000 fcec da89 715f 0000
        0x0040:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0050:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0060:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0070:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0080:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0090:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00a0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00b0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00c0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00d0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00e0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00f0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0100:  0000 0000 0000 0000 6382 5363 3501 013d
        0x0110:  0701 fcec da89 715f 3902 0240 3708 0103
        0x0120:  060c 0f1c 2a2b 3c04 7562 6e74 ff00 0000
        0x0130:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0140:  0000 0000 0000 0000
21:23:44.288449 bc:30:5b:e7:38:92 > fc:ec:da:89:71:5f, ethertype IPv4 (0x0800), length 324: (tos 0xc0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 310)
    192.168.57.1.bootps > 192.168.57.2.bootpc: [udp sum ok] BOOTP/DHCP, Reply, length 282, xid 0x1e35286a, Flags [none] (0x0000)
          Your-IP 192.168.57.2
          Client-Ethernet-Address fc:ec:da:89:71:5f
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Offer
            Lease-Time (51), length 4: 3600
            Subnet-Mask (1), length 4: 255.255.255.0
            Default-Gateway (3), length 4: 192.168.57.1
            Domain-Name (15), length 4: "test"
            Vendor-Option (43), length 6: 1.4.192.168.56.188
            Server-ID (54), length 4: 192.168.57.1
        0x0000:  45c0 0136 0000 0000 4011 85a3 c0a8 3901
        0x0010:  c0a8 3902 0043 0044 0122 c73d 0201 0600
        0x0020:  1e35 286a 0000 0000 0000 0000 c0a8 3902
        0x0030:  0000 0000 0000 0000 fcec da89 715f 0000
        0x0040:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0050:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0060:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0070:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0080:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0090:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00a0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00b0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00c0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00d0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00e0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00f0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0100:  0000 0000 0000 0000 6382 5363 3501 0233
        0x0110:  0400 000e 1001 04ff ffff 0003 04c0 a839
        0x0120:  010f 0474 6573 742b 0601 04c0 a838 bc36
        0x0130:  04c0 a839 01ff
21:23:44.289792 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x1e35286a, secs 3, Flags [none] (0x0000)
          Client-Ethernet-Address fc:ec:da:89:71:5f
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Request
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f
            Requested-IP (50), length 4: 192.168.57.2
            Server-ID (54), length 4: 192.168.57.1
            MSZ (57), length 2: 576
            Parameter-Request (55), length 8:
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)
            Vendor-Class (60), length 4: "ubnt"
        0x0000:  4500 0148 0000 0000 4011 79a6 0000 0000
        0x0010:  ffff ffff 0044 0043 0134 fab0 0101 0600
        0x0020:  1e35 286a 0003 0000 0000 0000 0000 0000
        0x0030:  0000 0000 0000 0000 fcec da89 715f 0000
        0x0040:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0050:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0060:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0070:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0080:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0090:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00a0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00b0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00c0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00d0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00e0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00f0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0100:  0000 0000 0000 0000 6382 5363 3501 033d
        0x0110:  0701 fcec da89 715f 3204 c0a8 3902 3604
        0x0120:  c0a8 3901 3902 0240 3708 0103 060c 0f1c
        0x0130:  2a2b 3c04 7562 6e74 ff00 0000 0000 0000
        0x0140:  0000 0000 0000 0000
21:23:47.293283 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x1e35286a, secs 6, Flags [none] (0x0000)
          Client-Ethernet-Address fc:ec:da:89:71:5f
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Request
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f
            Requested-IP (50), length 4: 192.168.57.2
            Server-ID (54), length 4: 192.168.57.1
            MSZ (57), length 2: 576
            Parameter-Request (55), length 8:
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)
            Vendor-Class (60), length 4: "ubnt"
        0x0000:  4500 0148 0000 0000 4011 79a6 0000 0000
        0x0010:  ffff ffff 0044 0043 0134 faad 0101 0600
        0x0020:  1e35 286a 0006 0000 0000 0000 0000 0000
        0x0030:  0000 0000 0000 0000 fcec da89 715f 0000
        0x0040:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0050:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0060:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0070:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0080:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0090:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00a0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00b0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00c0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00d0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00e0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00f0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0100:  0000 0000 0000 0000 6382 5363 3501 033d
        0x0110:  0701 fcec da89 715f 3204 c0a8 3902 3604
        0x0120:  c0a8 3901 3902 0240 3708 0103 060c 0f1c
        0x0130:  2a2b 3c04 7562 6e74 ff00 0000 0000 0000
        0x0140:  0000 0000 0000 0000
21:23:50.296357 fc:ec:da:89:71:5f > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x0, ttl 64, id 0, offset 0, flags [none], proto UDP (17), length 328)
    0.0.0.0.bootpc > 255.255.255.255.bootps: [udp sum ok] BOOTP/DHCP, Request from fc:ec:da:89:71:5f, length 300, xid 0x1e35286a, secs 9, Flags [none] (0x0000)
          Client-Ethernet-Address fc:ec:da:89:71:5f
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Request
            Client-ID (61), length 7: ether fc:ec:da:89:71:5f
            Requested-IP (50), length 4: 192.168.57.2
            Server-ID (54), length 4: 192.168.57.1
            MSZ (57), length 2: 576
            Parameter-Request (55), length 8:
              Subnet-Mask (1), Default-Gateway (3), Domain-Name-Server (6), Hostname (12)
              Domain-Name (15), BR (28), NTP (42), Vendor-Option (43)
            Vendor-Class (60), length 4: "ubnt"
        0x0000:  4500 0148 0000 0000 4011 79a6 0000 0000
        0x0010:  ffff ffff 0044 0043 0134 faaa 0101 0600
        0x0020:  1e35 286a 0009 0000 0000 0000 0000 0000
        0x0030:  0000 0000 0000 0000 fcec da89 715f 0000
        0x0040:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0050:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0060:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0070:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0080:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0090:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00a0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00b0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00c0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00d0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00e0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x00f0:  0000 0000 0000 0000 0000 0000 0000 0000
        0x0100:  0000 0000 0000 0000 6382 5363 3501 033d
        0x0110:  0701 fcec da89 715f 3204 c0a8 3902 3604
        0x0120:  c0a8 3901 3902 0240 3708 0103 060c 0f1c
        0x0130:  2a2b 3c04 7562 6e74 ff00 0000 0000 0000
        0x0140:  0000 0000 0000 0000
```

**Comment by yuwata** [comment]
Thank you! I will take a look.

**Comment by yuwata** [comment]
Progress report: I found that with the current git HEAD, the packet is accepted and ACK is sent from the server.
I will investigate which commit fixed the issue.

**Comment by yuwata** [comment]
@tebrevik If possible, could you also try the current git HEAD?

**Comment by yuwata** [comment]
Current v250-stable branch (https://github.com/systemd/systemd-stable/commit/e69b2a3a69d472e887633162111ed2a45f317eb9) also works fine.

**Comment by yuwata** [comment]
Hm, v249.10 (which was released few hours ago) also works fine.

**Comment by yuwata** [comment]
There is no critical difference between v249.9 and v249.10 in sd-dhcp-server.c...

**Comment by yuwata** [comment]
@tebrevik Could you show the relevant .network file?

**Comment by tebrevik** [comment]
> @tebrevik Could you show the relevant .network file?

Here it is:
```
[Match]
Name=eno2

[Network]
DHCPServer=yes
Address=192.168.57.1/24

[DHCPServer]
ServerAddress=192.168.57.1/24
PoolOffset=20
PoolSize=230
DefaultLeaseTimeSec=3600
MaxLeaseTimeSec=7200
SendVendorOption=01:ipv4address:192.168.56.188
SendOption=15:string:test

[DHCPServerStaticLease]
MACAddress=fc:ec:da:89:71:5f
Address=192.168.57.2

[DHCPServerStaticLease]
MACAddress=f0:9f:c2:a3:1f:ea
Address=192.168.57.3

```

**Comment by yuwata** [comment]
Ah, this is a duplicate of #20341. It is already fixed by b713a99b1a04507412b387214a26e43a2916bd01 (v250).

A workaround for the issue is to make the pool include the static lease.

Closing.
